src/main/java/org/gooru/insights/controllers/BaseController.java

```diff
@@ -58,13 +54,13 @@ public String getSessionToken(HttpServletRequest request) {
 	}
 	
 	public String getRequestData(HttpServletRequest request,String requestBody){
-		if(request.getMethod().equalsIgnoreCase(GET)){
-			if(request.getParameter(DATA) == null){
-				throw new BadRequestException(MessageHandler.getMessage(ErrorConstants.E100, DATA));
-			}
-			requestBody =  request.getParameter(DATA).toString();
-		}else if(requestBody == null){
-				throw new BadRequestException(MessageHandler.getMessage(ErrorConstants.E100, JSON_BODY));
+		
+		String data = request.getParameter(DATA);
+		if(data != null) {
+			requestBody = data;
+		}
```

Comment:
Can we use generic reject method to handle this kind of validation ? 
 and also 
Setter of data in request body should be done after validation.